tests/interpolation/test_custom_resolvers.py

```diff
@@ -345,44 +346,30 @@ def test_clear_cache(restore_resolvers: Any) -> None:
     assert old != c.k
 
 
-@mark.parametrize(
-    ("cfg", "expected"),
-    [
-        ({"a": 0, "b": 1}, {"a": 0, "b": 1}),
-        ({"a": "${y}"}, {"a": -1}),
-        ({"a": 0, "b": "${x.a}"}, {"a": 0, "b": 0}),
-        ({"a": 0, "b": "${.a}"}, {"a": 0, "b": 0}),
-        ({"a": "${..y}"}, {"a": -1}),
-    ],
-)
-def test_resolver_output_dict_to_dictconfig(
-    restore_resolvers: Any, cfg: Dict[str, Any], expected: Dict[str, Any]
-) -> None:
-    OmegaConf.register_new_resolver("dict", lambda: cfg)
+@mark.parametrize("readonly", [True, False])
+def test_resolver_output_dict(restore_resolvers: Any, readonly: bool) -> None:
+    some_dict = {"a": 0, "b": "${y}"}
+    OmegaConf.register_new_resolver("dict", lambda: some_dict)
     c = OmegaConf.create({"x": "${dict:}", "y": -1})
-    assert isinstance(c.x, DictConfig)
-    assert c.x == expected
-    assert dereference_node(c, "x")._get_flag("readonly")
-
-
-@mark.parametrize(
-    ("cfg", "expected"),
-    [
-        ([0, 1], [0, 1]),
-        (["${y}"], [-1]),
-        ([0, "${x.0}"], [0, 0]),
-        ([0, "${.0}"], [0, 0]),
-        (["${..y}"], [-1]),
-    ],
-)
-def test_resolver_output_list_to_listconfig(
-    restore_resolvers: Any, cfg: List[Any], expected: List[Any]
-) -> None:
-    OmegaConf.register_new_resolver("list", lambda: cfg)
+    OmegaConf.set_readonly(c, readonly)
+    assert isinstance(c.x, dict)
+    assert c.x == some_dict
+    x_node = dereference_node(c, "x")
+    assert isinstance(x_node, AnyNode)
+    assert x_node._get_flag("allow_objects")
+
+
+@mark.parametrize("readonly", [True, False])
+def test_resolver_output_list(restore_resolvers: Any, readonly: bool) -> None:
+    some_list = ["a", 0, "${y}"]
+    OmegaConf.register_new_resolver("list", lambda: some_list)
     c = OmegaConf.create({"x": "${list:}", "y": -1})
-    assert isinstance(c.x, ListConfig)
-    assert c.x == expected
-    assert dereference_node(c, "x")._get_flag("readonly")
+    OmegaConf.set_readonly(c, readonly)
+    assert isinstance(c.x, list)
+    assert c.x == some_list
+    x_node = dereference_node(c, "x")
+    assert isinstance(x_node, AnyNode)
+    assert x_node._get_flag("allow_objects")
```

Comment:
I think those two tests can be combined into one test using identity to pass through dict and list from the parameterization.

Reply:
Yes, good point, done in e4503d3 (I didn't use `identity` though because it would be a bit more cumbersome and harder to read IMO, it's easier to register a resolver that outputs the dict/list we want)